xchange-bitfinex/src/main/java/org/knowm/xchange/bitfinex/v2/dto/BitfinexExceptionV2.java

```diff
@@ -0,0 +1,24 @@
+package org.knowm.xchange.bitfinex.v2.dto;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import lombok.Getter;
+import lombok.ToString;
+
+@SuppressWarnings("serial")
+@Getter
+@ToString
+public class BitfinexExceptionV2 extends RuntimeException {
+
+  private String type;
+  private Long errorCode;
+  private String message;
+
+  // ["error",10100,"apikey: invalid"]
+  @JsonCreator
+  public BitfinexExceptionV2(Object[] arr) {
```

Comment:
can jackson handle `Object[]`? I have my doubts, shouldn't this be `String[]`?
could you provide a test using wiremock which shows a BitfinexExceptionV2 escaping a *Raw method?

Reply:
sure it can handle `Object[]`, see the comment over the method, this is, how an error could like passed by bitfinex `["error",10100,"apikey: invalid"]`

Reply:
then why does this happen in your code:

```
  public List<Status> getStatus(List<CurrencyPair> pairs) throws IOException {
    try {
      return bitfinexV2.getStatus(
          "deriv", BitfinexAdapters.adaptCurrencyPairsToTickersParam(pairs));
    } catch (HttpStatusIOException e) {
```

if BitfinexExceptionV2 works correctly then it should be thrown by `bitfinexV2.getStatus` instead of HttpStatusIOException

we can argue or we can implement a test that proves what happens ;)
you can find an example on how to write a test with wiremock here: https://github.com/knowm/XChange/blob/develop/xchange-livecoin/src/test/java/org/knowm/xchange/livecoin/service/TradeMockedIntegrationTest.java